x/incentives/types/msgs.go

```diff
@@ -138,3 +139,53 @@ func (m MsgAddToGauge) GetSigners() []sdk.AccAddress {
 	owner, _ := sdk.AccAddressFromBech32(m.Owner)
 	return []sdk.AccAddress{owner}
 }
+
+var _ sdk.Msg = &MsgCreateGroup{}
+
+// NewMsgCreateGroup creates a message to create a group with the provided parameters.
+func NewMsgCreateGroup(rewards sdk.Coins, numEpochsPaidOver uint64, owner sdk.AccAddress, poolIds []uint64) *MsgCreateGroup {
+	return &MsgCreateGroup{
+		Coins:             rewards,
+		NumEpochsPaidOver: numEpochsPaidOver,
+		Owner:             owner.String(),
+		PoolIds:           poolIds,
+	}
+}
+
+// Route takes a create group message, then returns the RouterKey.
+func (m MsgCreateGroup) Route() string { return RouterKey }
+
+// Type takes a create group message, then returns the message type.
+func (m MsgCreateGroup) Type() string { return TypeMsgCreateGroup }
+
+// ValidateBasic checks that the create group message is valid.
+func (m MsgCreateGroup) ValidateBasic() error {
```

Comment:
Should we have a constant bound on the max number of pools possible? E.g. 50 max.

Reply:
capped to 30 https://github.com/osmosis-labs/osmosis/pull/6588/commits/eaff91934d2443c963ba6dc18c4cc42e389b0c88